lib/matplotlib/quiver.py

```diff
@@ -487,6 +498,25 @@ def __init__(self, ax, *args,
         self.angles = angles
         self.width = width
 
+        # Checks the boundaries of mid_scale if outside range default of 0.5
+        if (head_pos == 0.0) or (head_pos == "tail"):
```

Comment:
There's no need to check both 0.0 and 'tail'. Instead, check for 'tail' (and the other strings), and set `head_pos = 0.0` as appropriate. Then do a final check for `0.0 <= head_pos <= 1.0`.